### PR TITLE
Add example to AngleEmbedding

### PR DESCRIPTION
**Pull request by ankit27kh**
Add example in documentation

### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `doc/releases/changelog-dev.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The PennyLane source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/).
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint pennylane/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Some templates do not have examples. 
**Description of the Change:**
Add example to AngleEmbedding documentation
**Benefits:**
Better documentation
**Possible Drawbacks:**
N/A
**Related GitHub Issues:**
#1727 #1735

### CODE REVIEW

**Comment by ankit27kh** [comment]
Is this example acceptable? If yes, similar examples can be made in all templates with further details where necessary.

File: pennylane/templates/embeddings/angle.py [hunk 1]
```diff
@@ -43,6 +43,33 @@ class AngleEmbedding(Operation):
             with :math:`N\leq n`
         wires (Iterable): wires that the template acts on
         rotation (str): type of rotations used
+
+    Example:
+
+        Angle embedding encodes the features as rotaion angles of the specified rotation angle.
```
Comment (antalszava) [inline pennylane/templates/embeddings/angle.py:49]:
```suggestion
        Angle embedding encodes the features by using the specified rotation operation.
```

File: pennylane/templates/embeddings/angle.py [hunk 2]
```diff
@@ -43,6 +43,33 @@ class AngleEmbedding(Operation):
...
+
+        .. code-block:: python
+
+            import pennylane as qml
```
Comment (antalszava) [inline pennylane/templates/embeddings/angle.py:53]: We could potentially assume that the import was made by the user. This just makes the example shorter.
```suggestion
```

File: pennylane/templates/embeddings/angle.py [hunk 3]
```diff
@@ -43,6 +43,33 @@ class AngleEmbedding(Operation):
...
+
+            @qml.qnode(dev)
+            def circuit(feature_vector):
+                qml.AngleEmbedding(features=feature_vector, wires=range(3), rotation='Z')
+                qml.Hadamard(0)
+                return qml.probs(wires=range(3))
+
+            X = [1,2,3]
+
+        Here, we have also used rotation angles RZ. If not specified, RX is used as default.
```
Comment (antalszava) [inline pennylane/templates/embeddings/angle.py:65]: Adding these small references should make the rotations clickable :slightly_smiling_face: 
```suggestion
        Here, we have also used rotation angles :class:`RZ`. If not specified, :class:`RX` is used as default.
```

**Comment by antalszava** [comment]
Hi @ankit27kh, thank you for the addition! :blush: 

> Hi @antalszava! Is this example acceptable? If yes, similar examples can be made in all templates with further details where necessary.

Yep, this looks awesome! Left some minor comments, but such examples look great. Let me know if you'd like us to leave a review. Don't forget to update the changelog file! :slightly_smiling_face:

**Pull request merged**